core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -661,16 +744,29 @@ final class ShuffleBlockFetcherIterator(
       result match {
         case r @ SuccessFetchResult(blockId, mapIndex, address, size, buf, isNetworkReqDone) =>
           if (address != blockManager.blockManagerId) {
-            if (hostLocalBlocks.contains(blockId -> mapIndex)) {
+            if (pushBasedFetchHelper.isMergedLocal(address)) {
+              // It is a local merged block chunk
+              assert(blockId.isShuffleChunk)
+              shuffleMetrics.incLocalBlocksFetched(pushBasedFetchHelper.getNumberOfBlocksInChunk(
+                blockId.asInstanceOf[ShuffleBlockChunkId]))
```

Comment:
I think we should count the number of blocks in the chunk and add it to the metrics here. This keeps these metrics consistent. The number of blocks fetched shouldn't change because they were merged. Also, we may have validations later that number of blocks that were generated by the map tasks are equal to the number of blocks fetched locally + remotely.
Would like to know what others think as well here 
cc @mridulm @Victsm

Reply:
I agree with @Ngone51 - if batch fetch is treating it as a single fetch for a set of contagious blocks fetched, makes sense to keep behavior consistent with that (this is essentially similar to all blocks being contiguous and being fetched by a batch fetch).

We can add another metric to capture what is being incremented here.

Reply:
> We can add another metric to capture what is being incremented here.

We have another jira for push-based metrics SPARK-33573. We will add a bunch of push related metrics with that.

Reply:
Was looking at the metrics that we will add with SPARK-33573 and we will add metrics such as :
1. number of total chunks
2. number of total blocks in that chunks.
3. failed chunks

So we will need to derive the actual number of local/remote shuffle blocks  from all these metrics.  
I think the validations can also be done on the bytes of data produced by map stage and read by read stage so I am fine with making the change.

Reply:
I have changes this as well. PTAL